src/main/java/org/tc/controllers/CourseController.java

```diff
@@ -111,17 +136,18 @@ public ModelAndView updateGet(@PathVariable("id") int id) {
             method = RequestMethod.POST)
     public ModelAndView updatePost(@PathVariable("id") int id,
                                    @Valid
-                                   @ModelAttribute("course") Course course,
+                                   @ModelAttribute("course") CourseForm course,
                                    BindingResult bindingResult) {
+        ModelAndView mav = new ModelAndView("classpath:views/" + UPDATE_VIEW_NAME);
```

Comment:
1. You use "classpath:views/" in dozens of places, but it is still not a constant.
2. Concatenation here looks not very pretty